### PR TITLE
Pixi v8 compatibility

### PR DESCRIPTION
**Pull request by vitch**
This PR builds on #468 so that I can install it in my app locally...

Inside my app, setting the relevant `package.json` entry to:

```
"pixi-viewport": "git+https://github.com/vitch/pixi-viewport#pixi-v8",
```

Pulls in this branch and let's me continue upgrading my app to v8...

### CODE REVIEW

**Comment by vitch** [comment]
See also #481 if you have issues with `yarn` and this branch...

**Comment by oliver-io** [comment]
Any news on this?  I've been using the branch from @vitch due to compat issues with v8.

**Comment by AZ604** [comment]
yeah, please merge

**Comment by adam-savard** [comment]
I'll echo that we're also using this branch for our stuff due to incompatibility issues!

**Comment by spassvogel** [comment]
@davidfig what are your thoughts? can this be merged?

**Comment by ArthurTimofey** [comment]
Works great on our end with this branch, using the package.json git url for now but would be nice to merge, happy to help with anything in that regard, thanks for the great work.

File: docs/src/code.js [hunk 1]
```diff
@@ -120,17 +119,19 @@ function overlap(x, y) {
 
 function stars() {
     const stars = (_viewport.worldWidth * _viewport.worldHeight) / Math.pow(STAR_SIZE, 2) * 0.1
+    console.log('stars', stars);
```
Comment (spassvogel) [inline docs/src/code.js:122]: this should be removed?

File: docs/src/code.js [hunk 2]
```diff
@@ -194,30 +196,38 @@ function API() {
...
-        // }
-        // _viewport.dirty = false
+    _application = new Application()
+    _application.init({
+        backgroundAlpha: 0,
+        width: window.innerWidth,
+        height: window.innerHeight,
+        resolution: window.devicePixelRatio
+    }).then(() => {
+        console.log(_application)
```
Comment (spassvogel) [inline docs/src/code.js:206]:
```suggestion

```

File: docs/src/code.js (hunk 1 above)
Comment (vitch) [reply docs/src/code.js:122]: Seems so... They are from the initial commits from @Aeolun which I just moved onto this PR so I could add the commits which made it usable for me...

**Comment by davidfig** [comment]
Merged #483, which I think does the same thing. Let me know if there are additional needed changes.

**Comment by spassvogel** [comment]
> Merged #483, which I think does the same thing. Let me know if there are additional needed changes.

Yes in fact I think @TimAstier had a good point here?

https://github.com/davidfig/pixi-viewport/pull/468/files#r1514200010

**Comment by spassvogel** [comment]
also this commit f4cc203abd5901914a90b1578fe6eda3ec8aea48 might be good?

**Comment by muhajirdev** [comment]
how did you guys got it to work? 
image

On my side I am getting this error

**Pull request closed without merging**